### PR TITLE
Topic/mafia docs

### PR DESCRIPTION
**Pull request by erikd-ambiata**
Not documenting the whole thing, just the ones I needed to look at the source for.

### CODE REVIEW

File: mismi-s3/src/Mismi/S3/Commands.hs [hunk 1]
```diff
@@ -384,7 +384,7 @@ getObjectsRecursively (Address (Bucket b) (Key ky)) =
         else
           pure $ resp ^. A.lorsContents
 
--- Pair of list of prefixes and list of keys
+-- | Return a tuple of the directories and files at the provided S3 Address.
```
Comment (nhibberd) [inline mismi-s3/src/Mismi/S3/Commands.hs]: Can we leave the wording here as `prefixes` and `keys`, a lot easier to understand in terms of s3

File: mismi-s3-core/src/Mismi/S3/Core/Data.hs [hunk 2]
```diff
@@ -112,16 +112,19 @@ combineKey (Key p1) (Key p2) =
     then Key $ p1 <> p2
     else Key $ p1 <> "/" <> p2
 
+-- | @withKey f address@ : Replace the 'Key' part of an 'Address' with a new
+--   'Key' resulting from the application of function @f@ to the old 'Key'.
 withKey :: (Key -> Key) -> Address -> Address
 withKey f (Address b k) =
   Address b $ f k
 
+-- | Get the directory name for a given key (eg. dirname "\/foo\/bar" == "foo").
```
Comment (nhibberd) [inline mismi-s3-core/src/Mismi/S3/Core/Data.hs]: Prefix here instead of directory

**Comment by nhibberd** [comment]
:+1: on comments

**Comment by erikd-ambiata** [comment]
@nhibberd claims ownership of the `branches-8-0-autoscaling` test failure and suggested I merge. He'll look at that failure when he gets in.

**Pull request merged**